### PR TITLE
Readability improvements

### PR DESCRIPTION
**Pull request by braekevelt**
I refactored `src/lib/components/TraceUpdater.react.js` to be more readable, but I am unable to test if the component works using the code and instructions from this repository. Please verify that the functionality is not broken before merging.

### CODE REVIEW

File: src/lib/components/TraceUpdater.react.js [hunk 1]
```diff
@@ -17,96 +14,79 @@ let _prev_layout = null;
...
-                    }
-                }
-                let index_arr = mergedUpdateData['index'];
-                delete mergedUpdateData['index'];
-                Plotly.restyle(graphDiv, mergedUpdateData, index_arr);
-            }
+        /// VALIDATION ///
+
+        const { id, gdID, sequentialUpdate, updateData } = this.props;
+        if (!isArray(updateData) || !head(updateData.length)) throw Error(`Invalid updateData '${updateData}'`);
```
Comment (jonasvdd) [inline src/lib/components/TraceUpdater.react.js]: What is the expected behaviour of calling `head` on `updatedata`.**`length`**; 

I assume you intend to call `!head(updateData)`

File: src/lib/components/TraceUpdater.react.js [hunk 2]
```diff
@@ -17,96 +14,79 @@ let _prev_layout = null;
...
+        
+        const graphDiv = document?.getElementById(gdID)?.getElementsByClassName('js-plotly-plot')?.[0];
+        if (!isElement(graphDiv)) throw Error(`Invalid graphDiv 'graphDiv' for gdID '${gdID}'`);
+        
+        
+        /// HELPER FUNCTIONS ///
+
+        const plotlyRestyle = ({ index, ...update }) =>
```
Comment (jonasvdd) [inline src/lib/components/TraceUpdater.react.js]: I'm not that acquainted with JavaScript, but I assume that defining these const-functions within the method itself are as efficient as defining them outside of this class/function.

(My intuition: as these are **const**-functions, they will not be re-constructed on each `render`-cal", but preinitialised and are thus as efficient as defining them outside of the function?)

**Comment by jonasvdd** [comment]
Already way more readable ❤️‍🔥

File: src/lib/components/TraceUpdater.react.js (hunk 2 above)
Comment (braekevelt) [reply src/lib/components/TraceUpdater.react.js]: No, the const-functions are redefined on each call. You could move them outside the class to be more efficient, except when they use variables from the scope of the outer function.

File: src/lib/components/TraceUpdater.react.js (hunk 1 above)
Comment (braekevelt) [reply src/lib/components/TraceUpdater.react.js]: Ah yes, that’s a mistake.

**Comment by jvdd** [comment]
Nice work @braekevelt ! :ok_hand:

File: src/lib/components/TraceUpdater.react.js (hunk 2 above)
Comment (jonasvdd) [reply src/lib/components/TraceUpdater.react.js]: Ah, but this poses minor overhead, I suppose?
(for encapsulation reasons, I think it is indeed better to leave that code within the `render`-function)

Comment (braekevelt) [reply src/lib/components/TraceUpdater.react.js]: This should be tested, but I indeed do not think this is the bottleneck.

**Review by jonasvdd — APPROVED** [review]

**Pull request merged**